mass2/core/misc.py

```diff
@@ -21,20 +22,30 @@ def show(fig: plt.Figure | None = None) -> mo.Html:
     return mo.mpl.interactive(fig)
 
 
-def pickle_object(obj: Any, filename: str) -> None:
+def pickle_object(obj: Any, filename: str | Path) -> None:
     """Pickle the given object to the given filename using dill.
     Mass2 Recipe objects are compatible with `dill` but _not_ with the standard `pickle` module."""
     with open(filename, "wb") as file:
         dill.dump(obj, file)
 
 
-def unpickle_object(filename: str) -> Any:
+def unpickle_object(filename: str | Path) -> Any:
     """Unpickle an object from the given filename using dill."""
     with open(filename, "rb") as file:
         obj = dill.load(file)
         return obj
 
 
+def pickle_stream(obj: Any) -> bytes:
+    """Pickle an object to the given stream using dill."""
+    return dill.dumps(obj)
+
+
+def unpickle_stream(stream: str | bytes) -> Any:
```

Comment:
here it suggests that `stream` is a str or bytes.... isn't that a chunk of data? Stream implies somehow the data is not all in memory at once, but I think here it is all in memory at once.